pkg/aggregator/aggregator.go

```diff
@@ -307,6 +308,12 @@ func NewBufferedAggregator(s serializer.MetricSerializer, eventPlatformForwarder
 		})
 	}
 
+	configID := pkgconfigsetup.Datadog().GetString("config_id")
+	if configID == "" {
+		// Ensure we never report an empty config ID
+		configID = "default"
+	}
```

Comment:
@coignetp Could you add the default value in https://github.com/DataDog/datadog-agent/blob/main/pkg/config/setup/config.go instead? Something like:

```
config.BindEnvAndSetDefault("config_id", "default")
```

You probably need to declare config_id there anyway to avoid unknown config warnings.

That way, if `pkgconfigsetup.Datadog().GetString("config_id")` is called somewhere else, we also get the default value easily.

Reply:
The default value is already set here https://github.com/DataDog/datadog-agent/pull/33068/files#diff-a7b6619528f155e64cefe12cc4ba753eb7688c87fdf9d96e1052a12eeb519df5R1139

Reply:
@coignetp hum, why do we need:

```
	if configID == "" {
		// Ensure we never report an empty config ID
		configID = "default"
	}
```

then? we will never cross this code path right?

Reply:
It's more a future proof, in case there is a change in the default value

Reply:
> It's more a future proof, in case there is a change in the default value

hum, I don't see why that will help with possible future change in default value.
We could just change it in https://github.com/DataDog/datadog-agent/blob/main/pkg/config/setup/config.go as easily.

But I see downsides of declaring default in 2 places:

- can lead to subtle bugs if default changes in `/pkg/config/setup/config.go`
- Code in `if configID == "" {` is a kind of dead code, never executed right now